src/test/java/org/minions/devfund/damianvp/SquarelotronTest.java

```diff
@@ -0,0 +1,306 @@
+        assertEquals(sevenTeen, se.getSquarelotron()[1][1]);
+        assertEquals(thirteen, se.getSquarelotron()[2][2]);
+        assertEquals(nine, se.getSquarelotron()[three][three]);
+        assertEquals(five, se.getSquarelotron()[four][four]);
+    }
+
+    /**
+     * test rotate rigth 6 turn.
+     */
+    @Test
+    public void testRotateRight6() {
+        final int three = 3;
+        final int six = 6;
+        final int fifteen = 15;
+        final int sixteen = 16;
+        final int twentyOne = 21;
+        final int twentyTwo = 22;
+        Squarelotron se = new Squarelotron(six);
+        se.rotateRight(six);
+        assertEquals(twentyTwo, se.getSquarelotron()[2][2]);
+        assertEquals(twentyOne, se.getSquarelotron()[2][three]);
+        assertEquals(sixteen, se.getSquarelotron()[three][2]);
+        assertEquals(fifteen, se.getSquarelotron()[three][three]);
+    }
+
+    /**
+     * test of diagonal flip. original squarelotron not should change.
+     */
+    @Test
+    public void testMainDiagonalFlipNewObject() {
+        final int four = 4;
+        final int five = 5;
+        final int seven = 7;
+        final int fifteen = 15;
+        final int twentyOne = 21;
+        final int twentyThree = 23;
+        final int twentyFive = 25;
+        Squarelotron se = new Squarelotron(five);
+        Squarelotron se2 = se.mainDiagonalFlip(1);
+        assertEquals(1, se.getSquarelotron()[0][0]);
+        assertEquals(seven, se.getSquarelotron()[1][1]);
+        assertEquals(five, se.getSquarelotron()[0][four]);
+        assertEquals(twentyThree, se.getSquarelotron()[four][2]);
+        assertEquals(twentyFive, se.getSquarelotron()[four][four]);
+        assertEquals(fifteen, se.getSquarelotron()[2][four]);
+        assertEquals(1, se2.getSquarelotron()[0][0]);
+        assertEquals(seven, se2.getSquarelotron()[1][1]);
+        assertEquals(twentyOne, se2.getSquarelotron()[0][four]);
+        assertEquals(fifteen, se2.getSquarelotron()[four][2]);
+        assertEquals(twentyFive, se2.getSquarelotron()[four][four]);
+        assertEquals(twentyThree, se2.getSquarelotron()[2][four]);
+    }
+
+    /**
+     * test main diagonal flip, dimension 5 and ring 1.
+     */
+    @Test
+    public void testMainDiagonalFlipDimension5Ring1() {
+        final int four = 4;
+        final int five = 5;
+        final int seven = 7;
+        final int fifteen = 15;
+        final int twentyOne = 21;
+        final int twentyThree = 23;
+        final int twentyFive = 25;
+        Squarelotron se = new Squarelotron(five);
+        se = se.mainDiagonalFlip(1);
+        assertEquals(1, se.getSquarelotron()[0][0]);
+        assertEquals(seven, se.getSquarelotron()[1][1]);
+        assertEquals(twentyOne, se.getSquarelotron()[0][four]);
+        assertEquals(fifteen, se.getSquarelotron()[four][2]);
+        assertEquals(twentyFive, se.getSquarelotron()[four][four]);
+        assertEquals(twentyThree, se.getSquarelotron()[2][four]);
```

Comment:
Try to re-write your unit tests in this way
```java
        final int ring = 1;
        final int dimension = 4;
        final Squarelotron squarelotron = new Squarelotron(dimension);
        final int[][] expectedResult = {
                {1, 5, 9, 13},
                {2, 6, 7, 14},
                {3, 10, 11, 15},
                {4, 8, 12, 16}
        };
        Squarelotron actualResult = squarelotron.mainDiagonalFlip(ring);
        assertArrayEquals(expectedResult, actualResult.getSquarelotronMatrix());
```

You can reuse the tests done by your teammates